ast.ts

```diff
@@ -83,35 +94,58 @@ export const ASSIGNABLE_TAGS = ["id", "lookup"] as const;
  * Subset of Expr types which are valid as assign targets
  */
 export type Assignable<A> =
```

Comment:
Is there any way we could use the `a` field for the `Locations`? For example:

```
parse : string -> Expr<Location>
type-check: Expr<Location> -> Expr<[Type, Location]>
```

We could add new elements to the tuple each time.

Then we could do e.g. `{ a: [typ, ...expr.a], [ more fields here ]}` in the type checker or other similar contexts that just add a single element to the front. We could also do it with an object type in that position. But it seems like a lot to always carry around the location, and to have _another_ optional field that we could forget in many positions.

Reply:
Will do. 

Our initial though on adding location as another field was because we thought the location does not need to go through the decorating process. But you are right, it is more efficient and extendable to use the "a" tag. (We forgot to talk about this during your OH yesterday, since you mentioned this in lecture before).